rtCamp/nginx-helper
# Add option to allow preloading cache.

## Description
The aim of this PR is to allow addition of preloading the cache.

## Type
- New Feature

## Reference Issue
- https://github.com/rtCamp/nginx-helper/issues/47

## Testing
The code has been tested locally and is working as expected.

To Test the functionality - 
1. Go to Nginx Helper page in the Settings.
2. Now select the checkbox for Preload Cache and Save.
3. Once the page reloads all the pages and posts ( which are available in sitemap.xml) will be cached.
4. You can test any Page/Post to  determine if it is cached or not.
5. To determine if the Page is cached or not we can do the following  - 
  a. Check if the timestamp generated by nginx timestamp has the queries executed as 1.
  b. Manually check in Redis if there are keys starting with prefix set for nginx helper in admin page.
  c. The timestamp in the HTML changes during each refresh so it is not reliable to determine if the page is cached or not.
